pkg/metamask/server.go

```diff
@@ -0,0 +1,35 @@
+package metamask
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"github.com/semrush/zenrpc/v2"
+	"go.uber.org/zap"
+	"log"
+	"net/http"
+	"os"
+)
+
+func RunMetaMaskService(ctx context.Context, address string) error {
+	rpc := zenrpc.NewServer(zenrpc.Options{ExposeSMD: true})
+	rpc.Register("", MetaMask{}) // public
+	rpc.Use(zenrpc.Logger(log.New(os.Stderr, "", log.LstdFlags)))
+
+	http.Handle("/", rpc)
+
+	server := &http.Server{Addr: address, Handler: nil}
+
+	go func() {
+		zap.S().Info("shutting down metamask service...")
```

Comment:
You should wait for `ctx.Done`. Or you should write this line after `err := server.Shutdown(ctx)`.